stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/internal/Channel.kt

```diff
@@ -255,3 +256,36 @@ internal fun Channel.updateUsers(users: Map<String, User>): Channel {
         this
     }
 }
+
+@InternalStreamChatApi
+public fun Channel.mergeChannelFromEvent(that: Channel): Channel {
+    return copy(
+        name = that.name,
+        image = that.image,
+        hidden = that.hidden,
+        frozen = that.frozen,
+        team = that.team,
+        config = that.config,
+        extraData = that.extraData,
+        syncStatus = that.syncStatus,
+        hiddenMessagesBefore = that.hiddenMessagesBefore,
+        memberCount = that.memberCount,
+        members = that.members,
+        lastMessageAt = when (that.lastMessageAt.after(lastMessageAt)) {
+            true -> that.lastMessageAt
+            else -> this.lastMessageAt
+        },
+        createdAt = that.createdAt,
+        updatedAt = that.updatedAt,
+        deletedAt = that.deletedAt,
+        /* Do not merge (messages, watcherCount, watchers, read, ownCapabilities, membership, unreadCount) fields.
+        messages = that.messages,
+        watcherCount = that.watcherCount,
+        watchers = that.watchers,
+        read = that.read,
+        ownCapabilities = that.ownCapabilities,
+        membership = that.membership,
+        unreadCount = that.unreadCount,
+        */
+    )
+}
```

Comment:
### Fix 3
We merged only valid fields from `event.channel` objects to void calling `enrichChannelsWithCapabilities` in `BatchUpdate` which may override our channels